drivers/serial/uart_nrfx_uart.c

```diff
@@ -335,17 +335,29 @@ static int uart_nrfx_init(struct device *dev)
 		 "UART init failed. Cannot find %s",
 		 CONFIG_GPIO_NRF5_P0_DEV_NAME);
 
-	(void) gpio_pin_configure(gpio_dev,
-				  CONFIG_UART_0_NRF_TX_PIN,
-				  (GPIO_DIR_OUT | GPIO_PUD_PULL_UP));
-	(void) gpio_pin_configure(gpio_dev,
-				  CONFIG_UART_0_NRF_RX_PIN,
-				  (GPIO_DIR_IN));
+	gpio_pin_configure(gpio_dev,
+			   CONFIG_UART_0_NRF_TX_PIN,
+			   (GPIO_DIR_OUT | GPIO_PUD_PULL_UP));
+	gpio_pin_configure(gpio_dev,
+			   CONFIG_UART_0_NRF_RX_PIN,
+			   (GPIO_DIR_IN));
 
 	nrf_uart_txrx_pins_set(NRF_UART0,
 			       CONFIG_UART_0_NRF_TX_PIN,
 			       CONFIG_UART_0_NRF_RX_PIN);
 
+#ifdef CONFIG_UART_0_NRF_FLOW_CONTROL
+	gpio_pin_configure(gpio_dev,
+			   CONFIG_UART_0_NRF_RTS_PIN,
+			   (GPIO_DIR_OUT | GPIO_PUD_PULL_UP));
```

Comment:
I think this setting is needed so that when UART is powered down, the output level is revert to the default so as to keep the UART lines sane. May be the macro name may not signify this, but I think we need to set the output state of the GPIO correct before UART is powered on.

Reply:
Configuring this pin as output is OK but enabling pull-up for an output is pointless IMHO.

Reply:
Refering to gpio_nrf5.c,  the macro GPIO_PUD_PULL_UP is used to set the output value.

Reply:
But is this behavior defined somewhere in the API? I couldn't find such description.
If it is a specific feature of the current nrf5 driver, we should not rely on it. A pair of `gpio_pin_write` and `gpio_pin_configure` will be safer in such case.